OneSignalSDK/onesignal/src/main/java/com/onesignal/outcomes/model/OSOutcomeEvent.java

```diff
@@ -49,38 +68,25 @@ public float getWeight() {
         return weight;
     }
 
-    public JSONObject toJSONObject() {
+    public JSONObject toJSONObject() throws JSONException {
         JSONObject json = new JSONObject();
-
-        try {
-            json.put(SESSION, session);
-            json.put(NOTIFICATION_IDS, notificationIds);
-            json.put(OUTCOME_ID, name);
-            json.put(TIMESTAMP, timestamp);
-            json.put(WEIGHT, weight);
-        } catch (JSONException exception) {
-            OneSignal.Log(OneSignal.LOG_LEVEL.ERROR, "Generating OutcomeEvent toJSONObject ", exception);
-        }
-
+        json.put(SESSION, session);
+        json.put(NOTIFICATION_IDS, notificationIds);
+        json.put(OUTCOME_ID, name);
+        json.put(TIMESTAMP, timestamp);
+        json.put(WEIGHT, weight);
         return json;
     }
 
-    JSONObject toJSONObjectForMeasure() {
+    public JSONObject toJSONObjectForMeasure() throws JSONException {
```

Comment:
This forces a try catch outside of this method why not just leave the try-catch internal so we dont need to expose it

Reply:
idem